GEODE-6603: Create StoppableCountDownLatch unit tests

kirklund opened the pull request:
* Create StoppableCountDownLatchTest and fixup any issues uncovered by testing.
* Remove unused class StoppableCountDownOrUpLatch.
* Improve ExecutorServiceRule for StoppableCountDownLatchTest.

moleske commented on geode-core/src/test/java/org/apache/geode/internal/util/concurrent/StoppableCountDownLatchTest.java:
```diff
@@ -0,0 +1,307 @@
+
+    latch.countDown();
+
+    // returns true
```
probably not super informative

moleske commented on geode-core/src/test/java/org/apache/geode/internal/util/concurrent/StoppableCountDownLatchTest.java:
```diff
@@ -0,0 +1,307 @@
+    Future<Boolean> latchFuture =
+        executorServiceRule.submit(() -> latch.await(theTimeoutMillis, MILLISECONDS));
+
+    // returns false
```
same dealio as above

kirklund commented:
@upthewaterspout We changed StoppableCountDownLatch to delegate to CountDownLatch instead of extending it and changed the ticket to "StoppableCountDownLatch should have unit tests"
